cert/tls.go

```diff
@@ -24,6 +26,28 @@ var (
 	}
 )
 
+// TlsReloader updates the TLS certificate without restarting the server.
+type TlsReloader struct {
```

Comment:
We should unify the capitalization of TLS everywhere to always be upper case (except for variables that start with the acronym, for example `tlsCfg`). So `TLSReloader`, `TestTLSConfig` and so on.

Reply:
Ah, and also please move the definition of the `struct` to the rest of the struct's definitions. I usually go with the order "struct definition, newXXX function, struct methods".